### ISSUE TITLE
Finalization of preCICE fails when coupled simulation finalizes

### ISSUE THREAD

**Issue by DavidSCN** [precice/su2-adapter#24]
As discussed on [discourse](https://precice.discourse.group/t/su2-failing-finalization/512), the following error might occur when the simulation finalizes

```bash
ERROR: The required actions write-iteration-checkpoint are not fulfilled. Did you forget to call “markActionFulfilled”?
```

There is probably something wrong when `isCouplingOngoing` returns false and we leave the time loop.

**Comment by BenjaminRodenberg** [comment]
I just ran the perpendicular flap with FEniCS and SU2 and also ran into this error at the very end.

**Comment by DavidSCN** [comment]
Closed via #25
